### DISCUSSION TITLE
Indentation via Tab and Shift+Tab should add whitespace

### DISCUSSION THREAD

**Discussion by Nantris** [facebook/lexical#4354] [Ideas]
I really think anything mapped to <kbd>Tab/Shift+Tab</kbd> outside of a list should be adding whitespace characters to indent/unindent, instead of `padding`.

The idea of padding-based indents is totally normal but should never be mapped to <kbd>Tab/Shift+Tab</kbd> in my opinion. These should be reserved for adding and removing `tab`-like whitespace. This is especially problematic because the current indentation method cannot be reflected in the exported HTML and to make it work would require external CSS be shipped along with the exported HTML.

Is there any openness to this suggestion among the development team? Or can anyone explain why it's handled this way? It really harms portability and goes against a user's expectation of what the tab buttons do.

In Microsoft Word, which is the closest thing in terms of handling indenting via styling instead of via inserting whitespace, the functionality is never mapped to <kbd>Tab/Shift+Tab</kbd> - and as far as I know uses <kbd>Ctrl+M/Ctrl+Shift+m</kbd> to show the formatting options instead.

**Comment by CanRau** [comment]
Very interesting observation, also curious although we're not (yet) supporting indentation in our editor 😇

**Comment by ashiq-prospero** [comment]
this should work as you expected

```
import { TabIndentationPlugin } from '@lexical/react/LexicalTabIndentationPlugin';
<TabIndentationPlugin maxIndent={10} />
```
